src/poc/miner_onion_server.erl

```diff
@@ -326,22 +318,53 @@ wait_until_next_block() ->
             ok
     end.
 
+wait_for_block(Fun, Count) when Count > 0 ->
+    ok = blockchain_event:add_handler(self()),
+    wait_for_block_(Fun, Count).
```

Comment:
should we unregister after this completes?

Reply:
it's only called in a spawn, is the event system going to unregister it automatically after it exits?